lib/src/game/camera.dart

```diff
@@ -0,0 +1,289 @@
+import 'dart:math' as math;
+import 'dart:ui' show Rect;
+
+import '../../components.dart';
+import '../../game.dart';
+
+/// Utility method to smoothly transition vectors.
+void _moveToTarget(
+  Vector2 current,
+  Vector2 target,
+  double ds,
+) {
+  if (current != target) {
+    final diff = target - current;
+    if (diff.length < ds) {
+      current.setFrom(target);
+    } else {
+      diff.scaleTo(ds);
+      current.setFrom(current + diff);
+    }
+  }
+}
+
+/// A camera translates your game coordinate system; this is useful when your
+/// world is not 1:1 with your screen size.
+///
+/// A camera always has a current [position], however you cannot set it
+/// directly. You must use some methods to ensure that the camera moves smoothly
+/// as the game runs. Smoothly here means that sudden snaps should be avoided,
+/// as they feel jarring to the player.
+///
+/// There are three major factors that determine the camera position:
+///
+/// * Follow
+/// If you want, you can call [followObject] at the beginning of your
```

Comment:
In the forge2d viewport the follow method can take some parameters to not follow the body strictly, so that your body/component can move around a bit in the middle without the camera having to move. I think that would be awesome to have, but that could be a follow-up too.
https://github.com/flame-engine/flame_forge2d/blob/master/lib/viewport.dart#L53

Reply:
I think what those parameters do is exactly the same as relativeOffset (the percentage of the screen the component occupies). we can definitely add more stuff as a followup though!